docs/src/userguide/language_basics.rst

```diff
@@ -760,12 +758,25 @@ An external C++ function that may raise an exception can be declared with::
 
 See :ref:`wrapping-cplusplus` for more details.
 
+Finally, if you are certain that your function cannot raise an exception, (e.g., it
+does not use Python objects at all), you can declare it as such using ``noexcept``::
```

Comment:
```suggestion
Finally, if you are certain that your function should not raise an exception, (e.g., it
does not use Python objects at all, or you plan to use it as a callback in C code that
is unaware of Python exceptions), you can declare it as such using ``noexcept``::
```

I just thought it worth mentioning callbacks, as something where ``noexcept`` probably
is the right thing to do.